### DISCUSSION TITLE
Chain Lightning no longer wakes its own creatures from Blindness

### DISCUSSION THREAD

**Discussion by PiotrKolobrzeg** [ihhub/fheroes2#9165] [Ideas]
### Preliminary checks

- [X] I've checked that there aren't [**other open issues**](https://github.com/ihhub/fheroes2/issues?q=is%3Aissue) on the same topic.
- [X] I've checked that this issue is reproducible on the [**latest snapshot build**](https://ihhub.github.io/fheroes2/INSTALL.html#snapshots-latest-builds).

### Platform

Windows

### Describe the bug

When an opponent uses Blind on your monster and you use Chain Lightning so that it wakes up from Blindness with its impact, the spell disappears, but the monster does not return to the turn and is not taken into account for that turn. This is a bug.

In the case of using Dispel Magic or Heal, the monster returns to the turn.

Please fix this bug.

Best regards

Piotr Rzepka

### Save file

No save file needed.

**Comment by zenseii** [comment]
@PiotrKolobrzeg. Have you confirmed this behavior with the original game? I know that if your blinded creature is struck by an enemy's attack spell, like magic arrow, then your creature does not get to do its turn that round. In fact this is an important tactic in the game.

**Comment by oleg-derevenetz** [comment]
> Have you confirmed this behavior with the original game?

Apparently not. I just checked it out, and in the original game, the unit skips a turn in the same way after any damage, including magic damage:

https://github.com/user-attachments/assets/a90bc768-88d9-45fe-a6da-0f14c6feca49

The blinded Cavalry unit does not get a turn, even though it is no longer blinded. So this is not a bug.

**Comment by zenseii** [comment]
As verified by @oleg-derevenetz, a blinded creature never gets it's turn back if it is attacked by a spell, friendly or otherwise.

We can close this issue.

Any suggestion to change this behavior should take place in a discussion first.

**Comment by PiotrKolobrzeg** [comment]
> @PiotrKolobrzeg. Have you confirmed this behavior with the original game? I know that if your blinded creature is struck by an enemy's attack spell, like magic arrow, then your creature does not get to do its turn that round. In fact this is an important tactic in the game.

@zenseii Hi, I didn't check it in the original game. The problem is with the latest version of fheroes2

This isn't about an enemy spell that can wake a unit from blindness. This is about my own chain lightning spell that I cast to wake my unit up, and it does wake up, but it can't move this turn anyway and doesn't return to the queue.

> Apparently not. I just checked it out, and in the original game, the unit skips a turn in the same way after any damage, including magic damage. The blinded Cavalry unit does not get a turn, even though it is no longer blinded. So this is not a bug.

@oleg-derevenetz This is exactly what you showed in the video. That's why I wrote that the video is unnecessary here, because everyone understands what it's about. In my opinion, this is a game bug that should be fixed. In my opinion, it should be assumed that this is a bug in the original game, a lack of polish.

> As verified by @oleg-derevenetz, a blinded creature never gets it's turn back if it is attacked by a spell, friendly or otherwise.
> Any suggestion to change this behavior should take place in a discussion first.

So @zenseii let's bring something like this up in the discussion.

**Reply by zenseii** [reply]
I've converted this into a discussion now.

**Comment by oleg-derevenetz** [comment]
> In my opinion, this is a game bug that should be fixed. In my opinion, it should be assumed that this is a bug in the original game, a lack of polish.

We don't regard some behavior of the original game as a bug just because someone doesn't like it.

**Comment by zenseii** [comment]
In my opinion, it is more consistent if a blinded creature always loses its turn when struck by any offensive spell (chain lightning, fireball, armageddon etc.), because I don't see why this should change whether it was cast by a friendly hero or an enemy hero.

In addition, changing this would devalue spells like dispel and cure as they wouldn't be as valueable to get if you know you can cast something like the relatively weak death ripple, or as the topic creator showed: the last chain lightning blow, and get more or less the same effect at the cost of (very little) damage. In other words, more spells that restore a creature's turn and remove blind would then exist.

**Reply by PiotrKolobrzeg** [reply]
> In my opinion, it is more consistent if a blinded creature always loses its turn when struck by any offensive spell (chain lightning, fireball, armageddon etc.), because I don't see why this should change whether it was cast by a friendly hero or an enemy hero.
> 
> In addition, changing this would devalue spells like dispel and cure as they wouldn't be as valueable to get if you know you can cast something like the relatively weak death ripple, or as the topic creature showed the last chain lightning blow, and get the more or less the same effect at the cost of very little damage. In other words, more spells that restore a creature's turn and remove blind would then exist.

I understand, this is some voice of reason. But for logic, pure logic, simply whenever a unit has the blindness removed, i.e. the spell is gone, it should regain the ability to move. Either way, you're right, it would weaken the importance of dispel and heal.

**Reply by Branikolog** [reply]
Hi!
From a logic perspective blindness could be healed with a cure. A blow on your head won't help you to see again. :) (Btw, not only attacking spells, but all physical attacks have same behaviour regarding blind spell.)
I bet you just got used to the blind logic of other Heroes game. But this is not the reason to change the original logic, that players are got used to for more than 25 years.

**Comment by PiotrKolobrzeg** [comment]
@Branikolog Hey, well, I'm new to this forum, you are experienced and probably your wisdom and reasoning are more purposeful. Anyway, I'm glad that I could confront the idea with you. Maybe in the future, another idea of ​​mine will be better. Best regards, Piotr
